pkg/ebpf/telemetry/docs/ebpf_instrumentation.md

```diff
@@ -0,0 +1,16 @@
+# EBPF Instrumentation
```

Comment:
Can we refer to this as "eBPF Helper Instrumentation"? We aren't instrumenting all of eBPF.

Reply:
The instrumentation is separate from helper telemetry though. It ties together because we use instrumentation to cache the telemetry struct. However, the instrumentation itself can be used for other things.